### PR TITLE
Apply home bookmark on page load

### PR DESCRIPTION
**Pull request by AdityaHegde**
- Clean up the dashboard url state to remove stale code.
- Add a dedicated wrapper component for cloud UI that applies home bookmark.

Note that the home bookmark is not applied if there is a url state or if there is any local user state for the dashbaord.

### CODE REVIEW

File: web-admin/src/features/dashboards/DashboardAdminStateProvider.svelte [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  const queryClient = useQueryClient();
+  $: bookmarks = getBookmarks(
+    queryClient,
+    $runtime?.instanceId,
+    $page.params.organization,
+    $page.params.project,
+    metricViewName,
+  );
+
+  function syncDashboardStateLocal() {
```
Comment (ericpgreen2) [inline web-admin/src/features/dashboards/DashboardAdminStateProvider.svelte:41]: It's a bit confusing to see "Local" is this function name given this function is called in the Cloud context. I can't tell what "Local" refers to.

File: web-common/src/features/dashboards/proto-state/dashboard-url-state.ts [hunk 2]
```diff
@@ -133,6 +118,16 @@ function gotoNewDashboardUrl(url: URL, newState: string, defaultState: string) {
...
 
+/**
+ * Pulls data from dashboard to create the url state.
+ * Any subsections in the future would be added here to build the final url state.
+ */
+export function useDashboardProto(ctx: StateManagers) {
+  return derived([ctx.dashboardStore], ([dashboard]) =>
+    getProtoFromDashboardState(dashboard),
+  );
+}
```
Comment (ericpgreen2) [inline web-common/src/features/dashboards/proto-state/dashboard-url-state.ts:129]: I like where this is headed

**Review by ericpgreen2 — APPROVED** [review]
Works as described. I left some ideas for code cleanup that you can consider or not before merging.

File: web-common/src/features/dashboards/stores/syncDashboardState.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+} from "@rilldata/web-common/runtime-client";
+import { get } from "svelte/store";
+
+export function syncDashboardState(
+  metricViewName: string,
+  metricsViewSpec: V1MetricsViewSpec | undefined,
+  metricsViewSchema: V1StructType | undefined,
+  timeRangeQuery: V1MetricsViewTimeRangeResponse | undefined,
+  preloadUrlState: string | null,
+) {
```
[the remark this answers is not in the document]
Comment (AdityaHegde) [reply web-common/src/features/dashboards/stores/syncDashboardState.ts:15]: Lets do a refactor in a future PR. Dont want to deal with regressions this close to the release.

**Pull request merged**